landlab/core/model_component.py

```diff
@@ -316,30 +376,43 @@ def var_loc(cls, name):
         str
             The location ('node', 'link', etc.) where a variable is defined.
         """
-        return cls._var_mapping[name]
+        return cls._info[name]["mapping"]
 
-    def initialize_output_fields(self):
+    def initialize_output_fields(self, dim=None):
```

Comment:
@mcflugen can you comment on whether you can think of a better name for `dim`. 

I made this to do some additional clean up to the route-to-multiple flow directions which use fields that are (n-grid-elements, max-number-of-receivers) in shape. 

I thought about `axis`, `second_dimension`... and don't like these, or any of the other ideas I came up with.

Reply:
@kbarnhart When I've done things like this in the past I've used something like *values_per_element*. For example, when data are stored in fields in an *xarray* *Dataset*, the dimensions of the array you describe would be something like *(node, receiver_nodes_per_node)*. What do you think?

Reply:
I like it. Have made the change.